src/traces/bar/defaults.js

```diff
@@ -33,7 +155,23 @@ module.exports = function supplyDefaults(traceIn, traceOut, defaultColor, layout
     coerce('base');
     coerce('offset');
     coerce('width');
-    coerce('text');
+
+    coerceArray('text', coerceString);
+
+    var textPosition = coerceArray('textposition', coerceEnumerated);
+
+    var hasBoth = Array.isArray(textPosition) || textPosition === 'auto',
+        hasInside = hasBoth || textPosition === 'inside',
+        hasOutside = hasBoth || textPosition === 'outside';
+    if(hasInside || hasOutside) {
+        var textFont = coerceArray('textfont', coerceFont, layout.font);
+        if(hasInside) {
+            coerceArray('insidetextfont', coerceFont, textFont, layout.font);
+        }
+        if(hasOutside) {
+            coerceArray('outsidetextfont', coerceFont, textFont, layout.font);
+        }
+    }
```

Comment:
We don't coerce entries inside data arrays or `arrayOk` attributes.

Non-text entries are usually via fallbacks later on in the pipeline e.g. in the drawing step for scatter `text` [here](https://github.com/plotly/plotly.js/blob/d3c59da02d4451842d56e5e933e0e13090af9171/src/components/drawing/index.js#L350).

The reason being: we try to make `Plots.supplyDefaults` scale ~ number of attribute as opposed to ~ length of the data.